sklearn/manifold/t_sne.py

```diff
@@ -767,10 +771,10 @@ def _fit(self, X, skip_num_points=0):
             pca = PCA(n_components=self.n_components, svd_solver='randomized',
                       random_state=random_state)
             X_embedded = pca.fit_transform(X)
-        elif isinstance(self.init, np.ndarray):
-            X_embedded = self.init
         elif self.init == 'random':
```

Comment:
Fixed this as well. I also wrote a test that generated this warning. Would it make sense to push this test? I'm not sure how to assert that *no* FutureWarning is generated (I found 'assert_no_warnings' but I see that it's depreciating).

The two lines currently in the test are:
`tsne = TSNE(init=np.zeros((2, 2)))`
` tsne.fit(np.array([[0], [0]]))`